spring-integration-core/src/main/java/org/springframework/integration/dsl/StandardIntegrationFlow.java

```diff
@@ -127,30 +124,29 @@ public void start() {
 
 	@Override
 	public void stop(Runnable callback) {
-		if (this.lifecycles.size() > 0) {
-			AggregatingCallback aggregatingCallback = new AggregatingCallback(this.lifecycles.size(), callback);
-			ListIterator<SmartLifecycle> iterator = this.lifecycles.listIterator(this.lifecycles.size());
-			while (iterator.hasPrevious()) {
```

Comment:
I didn't notice this, but stopping right to left was wrong; corrected in the new code.

Reply:
No, Gary, it was right. Because we started before from right to left and filled out the local `lifecycles` property.
Then we used this `lifecycles` in the `stop()` and therefore the first one in the flow was in the end of that list, so we again should iterate it backwards to stop in proper order.
Right now indeed we don't have an intermediate list therefore the logic is changed around a normal `integrationComponents` state.

Reply:
Ah... right.